ci-operator/step-registry/cucushift/installer/check/nutanix/vip/cucushift-installer-check-nutanix-vip-commands.sh

```diff
@@ -0,0 +1,102 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+check_result=0
+# Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+# to be able to SSH.
+if ! whoami &> /dev/null; then
+    if [[ -w /etc/passwd ]]; then
+        echo "${USER_NAME:-default}:x:$(id -u):0:${USER_NAME:-default} user:${HOME}:/sbin/nologin" >> /etc/passwd
+    else
+        echo "/etc/passwd is not writeable, and user matching this uid is not found."
+        exit 1
+    fi
+fi
+
+oc new-project vip-test
+oc new-app httpd -n vip-test
+oc expose svc/httpd
+oc scale deployment/httpd --replicas=3
+# Check all 3 pods was successfully deployed and running
+loop=10
+while [[ ${loop} -gt 0 ]]; do
+    ready_replicas=$(oc get deployment httpd -o jsonpath='{.status.readyReplicas}')
+    if [[ "${ready_replicas}" == 3 ]]; then
+        echo "Pass: check 3 pods was successfully deployed and running"
+        break
+    else
+        echo "Wait: not ready yet"
+        loop=$((loop - 1))
+        sleep 3
+    fi
+    if [[ ${loop} == 0 ]]; then
+        echo "Timeout: failed due to timeout"
+        check_result=$((check_result + 1))
+    fi
+done
+
+# Check round-robin algorithm, each pod was deployed on a different worker
+ep_count=$(oc get ep httpd -o jsonpath='{.subsets[0].addresses}' | jq length)
+if [[ "${ep_count}" == 3 ]]; then
+    echo "Pass: check each pod was deployed on a different worker"
+else
+    echo "Fail: check each pod was deployed on a different worker"
+    check_result=$((check_result + 1))
+fi
+
+# Check ingressVIP works well
+route_host=$(oc get routes httpd -o jsonpath='{.spec.host}')
```

Comment:
Per my understanding, the VIP setting in install-config.yaml should be shown in the ingressVIP and apiVIP url, so that we can confirm the setting takes effect, am I right?

Reply:
This case originate from https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-29434, according to my understand, it's checking ingress works with customer app
```
3. Check ingressVIP works well
$ oc get routes
Send ping request to the provided url a couple of times
$ ping <provided-url>
```
```
# oc get routes
NAME    HOST/PORT                                                 PATH   SERVICES   PORT       TERMINATION   WILDCARD
httpd   httpd-vip-test.apps.sgao-n2.qe.devcluster.openshift.com          httpd      8080-tcp                 None

# ping httpd-vip-test.apps.sgao-n2.qe.devcluster.openshift.com
PING httpd-vip-test.apps.sgao-n2.qe.devcluster.openshift.com (10.0.132.15) 56(84) bytes of data.
64 bytes from 10.0.132.15 (10.0.132.15): icmp_seq=1 ttl=255 time=177 ms
64 bytes from 10.0.132.15 (10.0.132.15): icmp_seq=2 ttl=255 time=174 ms

```

Reply:
The `10.0.132.15` is the ingress VIP setting in install-config.yaml, right?

Reply:
yes, so do you suggest we test ingressVIP(10.0.132.15) is in the ping response?

Reply:
For the case test point, I think we more care about the VIP setting in install-config.yaml take effect in the cluster, e.g: route url can be resolved to ingressVIP, api URL can be resolved to apiVIP, but not test if the apiURL and route URL can be accessed or not. That is not the point for the vip test case points.

Reply:
whatever using `ping` or `curl` tools, the first thing is to ensure the URL is resolved to the VIP that set in install-config.yaml, of course, we can run some simple testing together, ensure the URL can be accessed.